### PR TITLE
docs: recommend offline bcrypt (#10050)

### LINKED ISSUE #10050: Expose `bcrypt` as part of `argocd`
# Summary

If `bcrypt` is an important component of how `argocd` works, and argo-cd (the project) doesn't believe everyone has easy access to `bcrypt` (I'm willing to believe this, although at this point, it might be possible to argue that everyone has `python` or even access to `docker`), then `argocd` (the program) should offer `bcrypt` as a command that can be run to do whatever is needed to perform a `bcrypt`.

# Motivation

https://argo-cd.readthedocs.io/en/stable/faq/#i-forgot-the-admin-password-how-do-i-reset-it

> To change the password, edit the argocd-secret secret and update the admin.password field with a new bcrypt hash. You can use a site like https://www.browserling.com/tools/bcrypt to generate a new hash.

One should *never* suggest a website for something relating to credentials. It's always the wrong answer.

# Proposal

Add a `argocd bcrypt` that supports performing whatever action this series of steps imagines a user needs to do.

---

Disclaimer: At this point, I am not remotely certain these steps are meaningful as they certainly don't correspond to the structure of my instances.

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Partially fixes #10050.

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -43,8 +43,9 @@ per [the getting started guide](getting_started.md). For Argo CD v1.9 and later,
 a secret named `argocd-initial-admin-secret`.
 
 To change the password, edit the `argocd-secret` secret and update the `admin.password` field with a new bcrypt hash.
-You can use a site like [https://www.browserling.com/tools/bcrypt](https://www.browserling.com/tools/bcrypt) to generate
-a new hash. For example:
+(Use a trustworthy, offline `bcrypt` implementation such as the [Python bcrypt library](https://pypi.org/project/bcrypt/) to generate the hash.)
```
Comment (jsoref) [inline docs/faq.md:46]: Could you provide an example of using the python shell to encrypt `hello`?

**Review by jsoref — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]
LGTM!

**Review by ashutosh16 — APPROVED** [review]
LG

**Review by jannfis — APPROVED** [review]
LGTM.

On another note, I think it'd be a good addition to `argocd admin` to provide such a functionality. Maybe even a password reset/set function, instead of just producing a bcrypt hash. :)

**Pull request merged**